src/smc-hub/client.coffee

```diff
@@ -2239,6 +2239,15 @@ class exports.Client extends EventEmitter
                 return
             subscription = undefined
             async.series([
+                (cb) =>
+                    dbg("Update the subscription.")
+                    changes =
+                        quantity : mesg.quantity
+                        plan     : mesg.plan
+                        coupon   : mesg.coupon_id
+                    @_stripe.customers.updateSubscription(customer_id, subscription_id, changes, cb)
+                (cb) =>
+                    @database.stripe_update_customer(account_id : @account_id, stripe : @_stripe, customer_id : customer_id, cb: cb)
                 (cb) =>
                     if mesg.coupon_id
```

Comment:
There is no cb called at all if mesg.coupon_id isn't true!  Oops.  Make the code

```
if not mesg.coupon_id
    cb()
... then the rest not indented
```

Reply:
Fixed. Weird how this change is being recorded.

Reply:
You forgot the return after the cb(), which is even worse :-)